pkg/nsx/services/securitypolicy/builder.go

```diff
@@ -661,9 +665,9 @@ func (service *SecurityPolicyService) buildRuleDisplayName(rule *v1alpha1.Securi
 		// e.g. input: security policy's rule name: TCP.http_UDP.1234_ingress_allow,
 		// expand to NSX security policy rules with name TCP.http_UDP.1234_TCP.80_ingress_allow and TCP.http_UDP.1234_UDP.1234_ingress_allow.
 		// in case that user defined input security policy's rule name: sp_namedport_rule,
-		// expand to NSX security policy rules with name sp_namedport_rule.TCP.80_ingress_allow and sp_namedport_rule.UDP.1234_ingress_allow.
+		// expand to NSX security policy rules with name sp_namedport_rule_TCP.80_ingress_allow and sp_namedport_rule_UDP.1234_ingress_allow.
 		index := strings.Index(ruleName, common.ConnectorUnderline+suffix)
-		return util.GenerateTruncName(common.MaxNameLength, ruleName[:index]+"."+service.buildRulePortString(&rule.Ports[portIdx], true, portNumber), "", suffix, "", ""), nil
+		return util.GenerateTruncName(common.MaxNameLength, ruleName[:index]+common.ConnectorUnderline+service.buildRulePortString(&rule.Ports[portIdx], true, portNumber), "", suffix, "", ""), nil
```

Comment:
I prefer `sp_namedport_rule.UDP.1234_ingress_allow`, in which "rule.UDP.1234" is a description for the rule itself. If we switch to `sp_namedport_rule_UDP.1234_ingress_allow`, it is difficult to parse the rule details from "rule_UDP.1234".

Reply:
Good advice, revert the change back as before.